### DISCUSSION TITLE
Using page title as header?

### DISCUSSION THREAD

**Discussion by magpie514** [SilentVoid13/Templater#791] [Help]
I am not quite sure how to achieve this. I want to have a template that has a header equal to as the current file's name (as seen in the file tree, no ".md" or path or anything).
When I create a new note, I go to a folder in the file view, right click, new note, and then the cursor goes to the header bar where I can give the file a name. When I try this using "create new file from template", the note is _already generated at that point_, so it will remain "Untitled" no matter what. Unless I change it manually, which defeats the purpose.

What I want to achieve is to have the header get the same title as the text I input when Obsidian asks for a name. I don't know nor use any other method than the one described above to add notes to my vault (beginner user, doing nothing fancy).

Every other answer related to this seems to involve some sort of date automation to generate dynamic headers or filenames. I don't want that. I just want the new note to be have a header that matches the name I give it _from within_ obsidian.

Assumptions and details: 
* The name is not going to change. 
* It's not going to be dynamic. No variables, dates or anything involved.
* Path is irrelevant. 
* There is no scripting anywhere.
* If the file is renamed externally or anything, it's irrelevant. One-and-done deal here.
* This is not part of another larger operation I'm not mentioning. This is the end goal.

The purpose is merely to not have to type the note's title twice, that's it. I've read of other plugins to automatically scan and sync H1 headers to the filenames but that's honestly overkill and I never really change name or header once started on a note.

**Comment by AB1908** [comment]
Let me see if I understand this correctly. Your steps:

1. You are in a note "Sample.md".
2. You want to create a new note "Example.md" whose H1 is `Sample`.
3. Happily ever after?

**Reply by magpie514** [reply]
1. I'm creating a new note from nowhere.
2. I want to create a note, say "groceries". Normally I'd press the new note button and this happens, do note the cursor:
![image](https://user-images.githubusercontent.com/323651/188271378-6905061e-83c5-436e-9fd8-96f3899a128b.png)
3. I want `# <what I just wrote in the title bar>` as the resulting raw content so it's the H1.

Doing the same thing with "create new note from template" does not work, because the raw content is already generated. So that means I don't quite know how to achieve this simple thing, and all responses I see assume a much more complex task (that's why I tried to make clear this wasn't part of a more complex chain of actions, or a step in a script or something).

**Reply by petzi53** [reply]
I am not sure if I understood you correctly. An Obsidian community plugin synchronizes filename with heading and vice versa. This may solve your problem. [https://github.com/dvcrn/obsidian-filename-heading-sync](https://github.com/dvcrn/obsidian-filename-heading-sync).

I use this plugin all the time and have set up hotkeys for these actions: CTRL-SHIFT-F (macOS) pressed when the cursor is in the title field, synchronizes (creates) my H1 header with the title (= filename). With the Templater plugin, I have created a template that puts the cursor on the position I want to have my H1 header. (In my setup, this is after the YAML header.)

Put `# <% tp.file.cursor() %>` at the place where you want to have your H1 header. Now you can jump with return or tab directly from the title field to your H1 header.

If I change the header, I press CTRL-SHIFT-H to synchronize the title and filename of my note.

Hope this helps.

**Reply by magpie514** [reply]
Hmm, sadly it's not quite ideal. As I said I don't want the files to be tracked for filename changes, that'd be overkill and I don't intend to rename them or change the header. And if I like using templates (if I can get this to work, that is) I want to use them in my Android vault to save touch screen typing, and won't have a keyboard to invoke hotkeys there.

I just want a template that gives me 

```
 # Title given through default Obsidian means (the header bar on new file creation)
```
Because when I generate a new note, it's already named "Untitled". Every Templater tag I found when searching will **always** become "Untitled" (or a path to an Untitled.md file). Or in other words it'll be already done before Obsidian itself gives me a chance to name the file.

There was an issue involving this but it didn't quite get anywhere (for my case) and got closed, everyone had complex use cases with files with automatic names (no user input) or files generated externally, or scripting, but none about the basic use case of generating a new note with the title given through default means ("new note from template"). Other possible solutions were mentioned as not working and out of date, and the ones I tried myself just didn't change from "Untitled". That's why I listed things I am not attempting to do, since people seemed more invested in the complex scenarios. 
I guess I overdid it and made it more confusing, sorry about that.

**Reply by welpdx** [reply]
So what about:
1. At anywhere, run "Templater: create new note" -> select template
2. The template prompts user for text
3. The template then replaces: 1) filename from "untitled" to the text 2) add first headline to text

Alternatively:
1. Obsidian Create new Note. (Default is Untitled)
2. Type out a header
3. Run "Templater: Open Insert Template" -> Select template
4. Template searches for the text of the first header
5. Template changes title to new header


Is one of these what you are looking for?

**Reply by magpie514** [reply]
Oh yes. Perhaps a way to hijack the note creation process so I get prompted for a new note name first, which then can be applied to the filename and used for templating. I never liked the way Obsidian does it, I feel there should be a dialog for new notes, it'd make this kinda issue non-existent.
Can you help me achieve this? Since I was unable to get this going I still haven't learned even basic Templater tags beyond date formats and such.

**Reply by AB1908** [reply]
There are a couple examples of it in the discussions. Essentially, you need to get a prompt and then use that input to run `tp.file.create_new()`.
